include/RAJA/util/Layout.hpp

```diff
@@ -186,7 +186,11 @@ struct LayoutBase_impl<camp::idx_seq<RangeInts...>, IdxLin, StrideOneDim> {
   template <camp::idx_t N, typename Idx, typename... Indices>
   RAJA_INLINE RAJA_HOST_DEVICE void BoundsCheck(Idx idx, Indices... indices) const
   {
-    if(!(0<=idx && idx < sizes[N])) BoundsCheckError<N>(idx);
+    if(static_cast<long int>(sizes[N]) > 0 &&
```

Comment:
Shouldn't we be using a template parameter type instead of casting to a specific type explicitly?

Reply:
Why do we need `sizes[N] > 0`, shouldn't that be caught in the `0 <= idx < sizes[N]` test even if sizes[N] is zero or negative?

Reply:
The trouble comes from projections. If ``size[N] !> 0``, then we have a projection in that dimension and idx could by anything.